src/infra/main/template.yaml

```diff
@@ -1346,6 +1399,146 @@ Resources:
       AliasName: !Sub "alias/${AWS::StackName}-CloudWatchEncryptionKey"
       TargetKeyId: !GetAtt CloudWatchEncryptionKey.Arn
 
+  #
+  #  Canary Deployment CloudWatch Alarms
+  #
+
+  StatusRetrieverFunctionErrorsAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      ActionsEnabled: true
+      AlarmActions:
+        - !Ref AlertingSNSHighAlertNotificationTopicARN
+      AlarmDescription: "Canary Deployment: Errors returned from the StatusRetrieverFunction Lambda"
+      MetricName: Errors
+      Dimensions:
+        - Name: Resource
```

Comment:
is this dim necessary?

Reply:
Hmmm I believe so, looking at this > 
<img width="905" alt="Screenshot 2024-04-17 at 16 08 26" src="https://github.com/govuk-one-login/account-interventions-service/assets/116737136/a0b0d47c-06d2-464a-8cee-2764639bafaf">

Reply:
I would say so - my understanding is that the dimension with `live` at the end will actually point to the new version of the lambda during switch over? @MinaMahdi - is this covered in the documentation anywhere?

Reply:
Just to double check @TaherKapasi , that Resource property - the return values for `!Ref` in docs are > _It returns the resource name of the underlying Lambda function._ so right to say may need to change that to > `!Sub "${AWS::StackName}-InterventionsProcessorFunction` or okay to keep as it is?

Reply:
@TaherKapasi - @chiarapascucci-gds and I had a look into the dimension again and we found that we need to add an extra dimension to the Error metric Alarms -https://catalog.workshops.aws/complete-aws-sam/en-US/module-5-canaries/40-monitor - > `Name: ExecutedVersion` - This dimension will point to the newest version of the Lambda so that we can make sure that it is the reason that this alarm is going off instead of a later version.